Source/Math/ConvolveGeometry.h

```diff
@@ -461,8 +470,7 @@ class ConvolveGeometry final
             if (mapCount.size() > 1)
                 dimOut *= mapCount[i];
             else if (i == inputShape.GetRank() - 1)
-                dimOut *= mapCount[0];
-
+                dimOut = dimOutNoDilation*mapCount[0];
```

Comment:
My understanding:
this whole '**dimout *='** thing is a hacky way to make the dimension calculation work for both convolution and pooling. It is handcrafted in a way that **dimOut** is both the output size for some dimension, and used as indicator for conv/pooling and stored in some place unoccupied.
My changes did the right culculation in the 'real dimension' with dilation but break the formula for the 'indicator' dimension.
So for simplicity I used the old formula without dilation to calculate the old value and put there. **dimOutNoDilation** means use old formula without considering dilation.